stl/inc/algorithm

```diff
@@ -10131,6 +10167,20 @@ namespace ranges {
             _STL_INTERNAL_STATIC_ASSERT(sentinel_for<_Se2, _It2>);
             _STL_INTERNAL_STATIC_ASSERT(indirect_strict_weak_order<_Pr, projected<_It1, _Pj1>, projected<_It2, _Pj2>>);
 
+            using _Memcmp_classification_pred =
+                typename decltype(_Lex_compare_memcmp_classify(_First1, _First2, _Pred))::_Pred;
+            if constexpr (!same_as<_Memcmp_classification_pred,
+                              void> && sized_sentinel_for<_Se1, _It1> && sized_sentinel_for<_Se2, _It2> //
+                          && same_as<_Pj1, identity> && same_as<_Pj2, identity>) {

```

Comment:
This should be `_Memcmp_in_lexicographical_compare_is_safe`

Reply:
I don't want to make too big changes in this PR. The purpose of this PR is just to enable optimizations. You can file a separate  issue to track this later.